test/e2e/e2e_test.go

```diff
@@ -153,3 +255,238 @@ var _ = Describe("e2e control plane", func() {
+		if err != nil {
+			framework.Failf("Expected container %s running on %s error %v", ovnContainer, ovnWorkerNode, err)
+		}
+		haMode = false
+		// Attempt to retrieve the pod name that will run the external interface for e2e control-plane ha mode
+		if kubectlOut == "''" {
+			haMode = true
+			kubectlOut, err = framework.RunKubectl("get", "pods", ovnNsFlag, "-l", labelFlag, jsonFlag, fieldSelectorHaFlag)
+			if err != nil {
+				framework.Failf("Expected container %s running on %s error %v", ovnContainer, ovnHaWorkerNode, err)
+			}
+		}
+		// Fail the test if no pod is matched within the specified node
+		if kubectlOut == "''" {
+			framework.Failf("Unable to locate container %s on any known nodes", ovnContainer)
+		}
+	})
+
+	It("Should validate connectivity within a hybrid-sdn namespace of pods on separate worker nodes", func() {
+		var err error
+		var pingTarget string
+		getPodIPRetry := 10
+		command := []string{"bash", "-c", "sleep 20000"}
+
+		// HA CI mode runs a named set of nodes with a prefix of ovn-control-plane
+		if haMode {
+			By(fmt.Sprintf("Creating a container on node %s and verifying connectivity to a pod on node %s", ovnHaWorkerNode2, ovnHaWorkerNode))
+			framework.Logf("Detected a HA mode KIND environment")
+			dstPingPodHaName := "e2e-dst-ping-ha-pod"
+			contHaName := fmt.Sprintf("%s-container", dstPingPodHaName)
+			createGenericPod(f, dstPingPodHaName, contHaName, ovnHaWorkerNode, command)
+			// There is a condition somewhere with e2e WaitForPodNotPending that returns ready
+			// before calling for the IP address will succeed. This simply adds some retries.
+			// Operation will fail after retries are exhausted (the less resources the longer it will take).
+			for i := 1; i < getPodIPRetry; i++ {
+				pingTarget, err = getPodAddress(dstPingPodHaName, f.Namespace.Name)
+				if err != nil {
+					framework.Logf("Warning unable to query the test pod on node %s %v", ovnHaWorkerNode, err)
+				}
+				if pingTarget != "<no value>" {
+					framework.Logf("Destination ping target for %s is %s", dstPingPodHaName, pingTarget)
+					break
+				}
+				time.Sleep(time.Second * 3)
+				framework.Logf("Retry attempt %d to get pod IP from initializing pod %s", i, dstPingPodHaName)
+			}
+			// Fail the test if no address is ever retrieved
+			if pingTarget == "<no value>" {
+				framework.Failf("Warning: Failed to get an IP for target pod %s, test will fail", dstPingPodHaName)
+			}
+			// Fail the test if the destination IP is empty
+			if pingTarget == "" {
+				framework.Failf("Warning: Failed to get an IP for target pod %s, failing test", dstPingPodHaName)
+			}
+			// Spin up another pod that attempts to reach the previously started pod on separate nodes
+			framework.ExpectNoError(
+				checkConnectivityPingToHost(f, ovnHaWorkerNode2, "ha-hybrid-sdn-e2e-ping", pingTarget, ipv4PingCommand, 30))
+		}
+
+		// non-HA CI mode runs a named set of nodes with a prefix of ovn-worker
+		if !haMode {
```

Comment:
@nerdalert aren't hamode and !hamode blocks doing the same logic, just with different node names? Could you reduce this into a single block and just make the node names conditional variables?

Reply:
Consolidated the blocks. Ty for the review.